lib/src/io/js.dart

```diff
@@ -287,7 +287,7 @@ Future<Stream<WatchEvent>> watchDir(String path, {bool poll = false}) async {
 
     return completer.future;
   } else {
-    var subscription = await ParcelWatcher.subscribeFuture(path,
+    var subscription = await parcelWatcher!.subscribe(path,
```

Comment:
Consider swapping the order of these conditions so you can write `if (parcelWatcher case parcel when !poll) {` and avoid a non-null assertion.